src/status_im2/common/contacts/events_test.cljs

```diff
@@ -0,0 +1,59 @@
+(ns status-im2.common.contacts.events-test
+  (:require [cljs.test :refer-macros [deftest is run-tests]]
+            [status-im2.common.contacts.events :as core]))
+
+;; initial db
+(def init-db
+  {:networks/current-network "mainnet_rpc"
+   :networks/networks        {"mainnet_rpc"
+                              {:id     "mainnet_rpc"
+                               :config {:NetworkId 1}}}})
+
+;; new identity (:searching state)
+(defn search-db
+  [input]
+  {:contacts/new-identity {:input      input
+                           :public-key nil
+                           :ens-name   nil
+                           :state      :searching
+                           :error      nil}})
+
+(deftest search-empty-string
+  (let [input    ""
+        ;; the expected db after code-under-test completes
+        expected {:db init-db}
+        ;; the actual return value of code-under-test
+        actual   (core/set-new-identity {:db init-db} input)]
+    (is (= actual expected))))
+
+(deftest search-ens
+  (let [input    "esep"
+        clean    (fn [db]
+                   (-> db
+                       (assoc-in [:resolve-public-key-from-ens-name :on-success] nil)
+                       (assoc-in [:resolve-public-key-from-ens-name :on-error] nil)))
+        expected {:db                               (merge init-db (search-db input))
+                  :resolve-public-key-from-ens-name
+                  {:chain-id   1
+                   :ens-name   (str input ".stateofus.eth")
+                   :on-success nil
+                   :on-error   nil}}
+        actual   (core/set-new-identity {:db init-db} input)]
+    (is (= (clean actual) expected))))
+
+(deftest search-compressed-key
+  (let [input    "zQ3shWj4WaBdf2zYKCkXe6PHxDxNTzZyid1i75879Ue9cX9gA"
+        clean    (fn [db]
+                   (-> db
+                       (assoc-in [:decompress-public-key :on-success] nil)
+                       (assoc-in [:decompress-public-key :on-error] nil)))
+        expected {:db                    (merge init-db (search-db input))
+                  :decompress-public-key
+                  {:public-key input
+                   :on-success nil
+                   :on-error   nil}}
+        actual   (core/set-new-identity {:db init-db} input)]
+    (is (= (clean actual) expected))))
+
+(comment
```

Comment:
I think we shouldn't commit this `comment` macro. I run tests for the current namespace differently, and other devs probably do it in different ways too.

Reply:
removed